Bottom: 55964c6a8be5434112603bdbab0adfa37f682e25
Top:    e9b7748412f26de493855d399ba84e4c2700ff65
Author: scott snyder <sss@karma>
Date:   2020-02-03 10:05:29 -0500

Fix compilation warnings.


---

diff --git a/source/src/DDMarlinCED.cc b/source/src/DDMarlinCED.cc
index fe77cb3..4a028a4 100644
--- a/source/src/DDMarlinCED.cc
+++ b/source/src/DDMarlinCED.cc
@@ -471,7 +471,7 @@ void DDMarlinCED::drawDD4hepDetector( dd4hep::Detector& lcdd, bool _surfaces, St
   // some models might not have a SurfaceManager extension:
   dd4hep::rec::SurfaceManager* sM = 0 ;
   try{  sM = lcdd.extension<dd4hep::rec::SurfaceManager>();
-  } catch( std::runtime_error ) {
+  } catch( const std::runtime_error& ) {
     lcdd.apply( "InstallSurfaceManager",0,0);
     sM = lcdd.extension<dd4hep::rec::SurfaceManager>();
   }
diff --git a/source/src/WeightedPoints3D.cc b/source/src/WeightedPoints3D.cc
index 511d94d..6a41f59 100644
--- a/source/src/WeightedPoints3D.cc
+++ b/source/src/WeightedPoints3D.cc
@@ -751,8 +751,8 @@ void WeightedPoints3D::propagateVecErrorsCartesian(){
     Dmat[1][1] =  sin(phi)*cos(theta) ;
     Dmat[2][1] =  -sin(theta) ;
     double thph_dmat [2][3];
-    for ( int i =0 ; i < 3 ; i++ ) {
-      for ( int j=0 ; j < 2 ; j++ ) {
+    for ( int i =0 ; i < 2 ; i++ ) {
+      for ( int j=0 ; j < 3 ; j++ ) {
         thph_dmat [i][j] = 0. ;
         for (int k=0 ; k < 2 ; k++ ) {
           thph_dmat [i][j] += _theta_phi_cov[i][k][l]*Dmat[j][k];
